Lib/test/test_unittest/testmock/testpatch.py

```diff
@@ -1979,5 +1979,32 @@ def test(): pass
             test()
 
 
+    def test_method_patched_by_subclass(self):
+        @patch('%s.something' % __name__, 1)
+        class Foo:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(Foo().test(), 1)
```

Comment:
Good point. The decorator from the parent class should be applied before that of the child class, but by calling the wrapper from the child first before it calls the wrapper from the parent, the order became reversed, resulting in the `Bar().test()` also returning 1. I've fixed it by obtaining the calling chain from the `__wrapped__` attribute and applying the patchings from the calling chain in reverse order.

Reply:
Thanks to your suggested tests, I've added all but `test_method_patched_by_subclass2`, which seems unnecessary to me because the subclass is not doing anything there.